packages/components/src/DateTimePickers/DateTime/datetime-extraction.js

```diff
@@ -120,11 +123,13 @@ function extractPartsFromDateTime(datetime, options) {
 		return {
 			date: undefined,
 			time: undefined,
+			datetime: undefined,
 		};
 	}
 	return {
 		date: extractDateOnly(datetime, options),
 		time: extractTimeOnly(datetime, options),
+		datetime,
```

Comment:
put `datetime` in state, so we can do quick equality check when props.value changes.